hazelcast-client/src/main/java/com/hazelcast/client/impl/ClientLoggingService.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl;
+
+import com.hazelcast.instance.BuildInfo;
+import com.hazelcast.logging.AbstractLogger;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.LogEvent;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.logging.LoggerFactory;
+import com.hazelcast.util.ConstructorFunction;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.logging.Level;
+
+import static com.hazelcast.util.ConcurrencyUtil.getOrPutIfAbsent;
+
+public class ClientLoggingService {
```

Comment:
I see little reasons why a logging interface should be aware who is using it (client, server,...) 
Certainly the interface should be agnostic to its users. 

The current implementation depends on `MemberImpl`, but this dependency can be easily removed - it depends on it just to create a prefix. If you supply the prefix externally then you can fully re-use the implementation. Do I miss anything?